python/cudf/cudf/utils/dtypes.py

```diff
@@ -147,6 +147,15 @@ def is_categorical_dtype(obj):
     return pandas_dtype(obj).type is CategoricalDtypeType
 
 
+def is_list_dtype(obj):
+    return (
+        type(obj) is cudf.core.dtypes.ListDtype
+        or obj is cudf.core.dtypes.ListDtype
+        or type(obj) is cudf.core.column.ListColumn
+        or obj is cudf.core.column.ListColumn
+    )
```

Comment:
What about if `obj` is a `Series` or `Index` of type `ListDtype`?

Reply:
Extended `is_list_like` to return true for this case, and also for `dtype == "list"`